### ISSUE TITLE
[ENHANCEMENT] Ability to use different metrics for calculating forecasting errors

### ISSUE THREAD

**Issue by carlomazzaferro** [carlomazzaferro/scikit-hts#18]
**Is your proposed enhancement related to a problem? Please describe.**
Right now the metric used to calculate the forecasting error is hard coded as the MSE: https://github.com/carlomazzaferro/scikit-hts/blob/master/hts/model/base.py#L69

Providing the ability to use custom metrics would greatly improve the usability of the library, as well as tailoring it to different use-cases

**Describe the solution you'd like**
Being able to pass a string for a built-in metric, or a callable. In case of a callable, the user must ensure the metric is implemented correctly. Examples of possible custom metrics in documentation should be added as well.

**Describe alternatives you've considered**
None, this is the first formalisation of the issue

**Additional context**
Metrics that should be built in:
- Mean squared error (MSE)
- Root mean squared error (RMSE)
- Weighted RMSE (RMSSE according to the M5 competition: https://mk0mcompetitiont8ake.kinstacdn.com/wp-content/uploads/2020/02/M5-Competitors-Guide_Final-1.pdf)
- Mean Absolute Error (MAE) 
- Mean absolute percentage error (MAPE)

Ideally the implementation follows the same interface as sklearn metric's: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_squared_error.html

**Comment by vtoliveira** [comment]
I will try to work on that, and give feedback as I advance. So for custom metrics you are also thinking 
in adding a hts.metrics interface as well? Because I am aware that sklearn does not implement a lot of them, just basic ones.

So in my mind would be something like:

```python
from hts.metrics import root_mean_squared_scaled_error
```

Also, we have [this Hyndman's](http://datascienceassn.org/sites/default/files/Another%20Look%20at%20Measures%20of%20Forecast%20Accuracy.pdf) paper that discuss all metrics dealing with time series and the ones for intermittent data as well. (For what I have seen, almost all Hierarchical data follows some kind of intermittend demand)

**Comment by vtoliveira** [comment]
Another question that I was thinking of, considering the case implemented above, the metric should automatically calculated rmse for each node or should we leave it to the user?

Because we have two different cases:
 - First, user passing metric as string / callable, in base.py each series is calculated at time.
 - Second, user get model.predict and compare with a test dataset. And we would have here for each y_true, y_pred shape -> (n_samples, n_nodes)

**Comment by vtoliveira** [comment]
Forget about it, sklearn already implements in this way, you choose if you want multioutput or not, I will try to follow their guidelines and practice so we have a robust metric. I will start with rmsse, the one used in M5 current competition.
